### DISCUSSION TITLE
Data List - support for dynamic root as replacement for XPath

### DISCUSSION THREAD

**Discussion by arknu** [leekelleher/umbraco-contentment#381] [Question] [answered]
XPath is going away in v14. V13 introduces Dynamic Root as a replacement for XPath in the Multi Url Picker. 

It would be useful to have a new Data Source for the data list that also uses this feature. I haven't looked at the code yet, but I would expect to be able to reuse much of the code from the Multi Url Picker implementation.

**Comment by leekelleher** [comment]
@arknu 💯 It's on my radar, but I haven't started any prototyping with it yet, (meaning that it's up-for-grabs).

I think at first I'd do it as a separate data-source, e.g. "Umbraco Content by Dynamic Root", mostly to keep backwards compatibility of the original "Umbraco Content" data-source.  Then for v14, look to merge the two into a combined data-source.

**Reply by arknu** [reply]
Yeah, it should definitely be a separate datasource for now.
I might have a look at implementing it for v13, if I get the time.

**Comment by leekelleher** [comment, marked as answer]
FYI, I have a pull request to add Dynamic Root support the existing Umbraco Content data-source.

- #386

I opted to do this instead of a separate data-source, as it was the simpler solution moving forwards.
